controller/Controller.h

```diff
@@ -0,0 +1,188 @@
+  socket->initialize();
+  std::vector<uint8_t> data;
+  uint8_t op = minifi::c2::Operation::UPDATE;
+  minifi::io::BaseStream stream;
+  stream.writeData(&op, 1);
+  stream.writeUTF("flow");
+  stream.writeUTF(file);
+  socket->writeData(const_cast<uint8_t*>(stream.getBuffer()), stream.getSize());
+
+  // read the response
+  uint8_t resp = 0;
+  socket->readData(&resp, 1);
+  if (resp == minifi::c2::Operation::DESCRIBE) {
+    uint16_t connections = 0;
+    socket->read(connections);
+    out << connections << " are full" << std::endl;
+    for (int i = 0; i < connections; i++) {
+      std::string fullcomponent;
+      socket->readUTF(fullcomponent);
+      out << fullcomponent << " is full" << std::endl;
+    }
+  }
+}
+
+/**
+ * Lists connections which are full
+ * @param socket socket ptr
+ */
+void getFullConnections(std::unique_ptr<minifi::io::Socket> socket, std::ostream &out) {
+  socket->initialize();
+  std::vector<uint8_t> data;
+  uint8_t op = minifi::c2::Operation::DESCRIBE;
+  minifi::io::BaseStream stream;
+  stream.writeData(&op, 1);
+  stream.writeUTF("getfull");
+  socket->writeData(const_cast<uint8_t*>(stream.getBuffer()), stream.getSize());
+
+  // read the response
+  uint8_t resp = 0;
+  socket->readData(&resp, 1);
+  if (resp == minifi::c2::Operation::DESCRIBE) {
+    uint16_t connections = 0;
+    socket->read(connections);
+    out << connections << " are full" << std::endl;
+    for (int i = 0; i < connections; i++) {
+      std::string fullcomponent;
+      socket->readUTF(fullcomponent);
+      out << fullcomponent << " is full" << std::endl;
+    }
+
+  }
+}
+
+/**
+ * Prints the connection size for the provided connection.
+ * @param socket socket ptr
+ * @param connection connection whose size will be returned.
+ */
+void getConnectionSize(std::unique_ptr<minifi::io::Socket> socket, std::ostream &out, std::string connection) {
+  socket->initialize();
+  std::vector<uint8_t> data;
+  uint8_t op = minifi::c2::Operation::DESCRIBE;
+  minifi::io::BaseStream stream;
+  stream.writeData(&op, 1);
+  stream.writeUTF("queue");
+  stream.writeUTF(connection);
+  socket->writeData(const_cast<uint8_t*>(stream.getBuffer()), stream.getSize());
+
+  // read the response
+  uint8_t resp = 0;
+  socket->readData(&resp, 1);
+  if (resp == minifi::c2::Operation::DESCRIBE) {
+    std::string size;
+    socket->readUTF(size);
+    out << "Size/Max of " << connection << " " << size << std::endl;
+  }
+}
+
+void listProcessors(std::unique_ptr<minifi::io::Socket> socket, std::ostream &out) {
+  socket->initialize();
+  minifi::io::BaseStream stream;
+  uint8_t op = minifi::c2::Operation::DESCRIBE;
+  stream.writeData(&op, 1);
+  stream.writeUTF("processors");
+  socket->writeData(const_cast<uint8_t*>(stream.getBuffer()), stream.getSize());
+  uint16_t responses = 0;
+  socket->readData(&op, 1);
+  socket->read(responses);
+  out << "Processors:" << std::endl;
```

Comment:
Might be a nice add where I can specify a flag that strips out headers by default.  Would save an inline sed or similar